servers/zms/src/main/java/com/yahoo/athenz/zms/store/jdbc/JDBCConnection.java

```diff
@@ -4130,6 +4157,19 @@ private boolean updateRoleMemberNotificationTimestamp(String server, long timest
         return roles;
     }
 
+    // To avoid firing multiple queries against DB, this function will generate 1 consolidated query for all domains->templates combination
+    public String generateDomainTemplateVersionQuery(Map<String, Integer> templateNameAndLatestVersion) {
+        StringBuilder query = new StringBuilder("SELECT domain.name, domain_template.template FROM domain_template " +
+                "JOIN domain ON domain_template.domain_id=domain.domain_id WHERE ");
+
+        for (String templateName : templateNameAndLatestVersion.keySet()) {
+            query.append("(domain_template.template = '" + templateName + "' and current_version < " + templateNameAndLatestVersion.get(templateName) + ") OR ");
```

Comment:
why not just call multiple appends instead of adding the strings first and then calling append?

Reply:
Done.